static/app/views/automations/components/automationListTable/actions.tsx

```diff
@@ -0,0 +1,205 @@
+import {Fragment, useCallback, useState} from 'react';
+import styled from '@emotion/styled';
+
+import {openConfirmModal} from 'sentry/components/confirm';
+import {Alert} from 'sentry/components/core/alert';
+import {Button} from 'sentry/components/core/button';
+import {Checkbox} from 'sentry/components/core/checkbox';
+import {Flex} from 'sentry/components/core/layout';
+import {SimpleTable} from 'sentry/components/tables/simpleTable';
+import {t, tct, tn} from 'sentry/locale';
+import {decodeScalar} from 'sentry/utils/queryString';
+import useLocationQuery from 'sentry/utils/url/useLocationQuery';
+import usePageFilters from 'sentry/utils/usePageFilters';
+import {
+  useDeleteAutomationsMutation,
+  useUpdateAutomationsMutation,
+} from 'sentry/views/automations/hooks';
+
+interface AutomationsTableActionsProps {
+  allResultsVisible: boolean;
+  canDisable: boolean;
+  canEnable: boolean;
+  pageSelected: boolean;
+  queryCount: string;
+  selected: Set<string>;
+  togglePageSelected: (pageSelected: boolean) => void;
+}
+
+export function AutomationsTableActions({
+  selected,
+  pageSelected,
+  togglePageSelected,
+  queryCount,
+  allResultsVisible,
+  canEnable,
+  canDisable,
+}: AutomationsTableActionsProps) {
+  const [allInQuerySelected, setAllInQuerySelected] = useState(false);
+  const anySelected = selected.size > 0;
+
+  const {selection} = usePageFilters();
+  const {query} = useLocationQuery({
+    fields: {
+      query: decodeScalar,
+    },
+  });
+
+  const {mutateAsync: deleteAutomations, isPending: isDeleting} =
+    useDeleteAutomationsMutation();
+  const {mutateAsync: updateAutomations, isPending: isUpdating} =
+    useUpdateAutomationsMutation();
+
+  const getConfirmMessage = useCallback(
+    (action: string) => {
+      if (allInQuerySelected) {
+        return tct(
+          'Are you sure you want to [action] all [queryCount] automations that match the search?',
+          {
+            action,
+            queryCount,
+          }
+        );
+      }
+      return tn(
+        // Use sprintf argument swapping since the number value must come
+        // first. See https://github.com/alexei/sprintf.js#argument-swapping
+        `Are you sure you want to %2$s this %s automation?`,
+        `Are you sure you want to %2$s these %s automations?`,
+        selected.size,
+        action
+      );
+    },
+    [allInQuerySelected, queryCount, selected.size]
+  );
+
+  const handleUpdate = useCallback(
+    ({enabled}: {enabled: boolean}) => {
+      openConfirmModal({
+        message: getConfirmMessage(enabled ? t('enable') : t('disable')),
+        confirmText: enabled ? t('Enable') : t('Disable'),
+        priority: 'danger',
+        onConfirm: () => {
+          if (allInQuerySelected) {
+            updateAutomations({enabled, query, projects: selection.projects});
```

Comment:
If you want the modal to wait until the request is finished you'll need to return the async function. Otherwise the modal won't have anything to await